engine/table/src/main/java/io/deephaven/engine/table/impl/select/analyzers/SelectAndViewAnalyzer.java

```diff
@@ -519,195 +851,169 @@ public void close() {
      * @param jobScheduler scheduler for parallel sub-tasks
      * @param liveResultOwner {@link LivenessNode node} to be used to manage/unmanage results that happen to be
      *        {@link io.deephaven.engine.liveness.LivenessReferent liveness referents}
-     * @param onCompletion Called when an inner column is complete. The outer layer should pass the {@code onCompletion}
-     */
-    public abstract void applyUpdate(TableUpdate upstream, RowSet toClear, UpdateHelper helper,
-            JobScheduler jobScheduler, @Nullable LivenessNode liveResultOwner,
-            SelectLayerCompletionHandler onCompletion);
-
-    /**
-     * Our job here is to calculate the effects: a map from incoming column to a list of columns that it effects. We do
-     * this in two stages. In the first stage we create a map from column to (set of dependent columns). In the second
-     * stage we reverse that map.
+     * @param onSuccess called when the update completed successfully
+     * @param onError called when the update failed
      */
-    public final Map<String, String[]> calcEffects(boolean forcePublishAllResources) {
-        final Map<String, Set<String>> dependsOn = calcDependsOnRecurse(forcePublishAllResources);
-
-        // Now create effects, which is the inverse of dependsOn:
-        // An entry W -> [X, Y, Z] in effects means that W affects X, Y, and Z
-        final Map<String, List<String>> effects = new HashMap<>();
-        for (Map.Entry<String, Set<String>> entry : dependsOn.entrySet()) {
-            final String depender = entry.getKey();
-            for (final String dependee : entry.getValue()) {
-                effects.computeIfAbsent(dependee, dummy -> new ArrayList<>()).add(depender);
+    public void applyUpdate(
+            final TableUpdate upstream,
+            final RowSet toClear,
+            final UpdateHelper helper,
+            final JobScheduler jobScheduler,
+            @Nullable final LivenessNode liveResultOwner,
+            final Runnable onSuccess,
+            final Consumer<Exception> onError) {
+
+        remainingLayers.or(requiredLayers);
+
+        final Runnable[] runners = new Runnable[layers.length];
+        final UpdateScheduler scheduler = new UpdateScheduler(runners, onSuccess, onError);
+
+        for (int ii = 0; ii < layers.length; ++ii) {
+            final Layer layer = layers[ii];
+            if (layer != null) {
+                runners[ii] = layer.createUpdateHandler(
```

Comment:
Can we add a comment here:
// Right now, this will allow concurrent layers to fail without ensuring that other sub-tasks are finished. We should try to clean that up.